### PR TITLE
refactor: refactor routerBasePath code with context and provider to c…

### PR DESCRIPTION
**Pull request by bigbigDreamer**
refactor: refactor routerBasePath code with context and provider to consume

### CODE REVIEW

**Review by myNameIsDu — CHANGES_REQUESTED** [review]

File: packages/launcher/src/module/launcherProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { type FC, createContext } from 'react';
+
+export type LauncherContextProps = {
+    basename?: string;
+};
+
+export const LauncherContext = createContext<LauncherContextProps>({});
+
+export type LauncherProviderProps = {
+    value: Record<string, any>;
```
[the remark this answers is not in the document]
Comment (bigbigDreamer) [reply packages/launcher/src/module/launcherProvider.tsx:10]: 可以

File: packages/launcher/src/module/app.tsx [hunk 2]
```diff
@@ -83,12 +85,13 @@ class Launcher {
...
-                routerBasePath={routerBasePath}
-                store={store}
-                hash={hash}
-                routes={routes}
-            />,
+            <LauncherProvider
+                value={{
+                    basename: routerBasePath,
+                }}
+            >
```
Comment (myNameIsDu) [inline packages/launcher/src/module/app.tsx:92]: 要用useMemo包一下

File: packages/launcher/src/module/app.tsx [hunk 3]
```diff
@@ -50,6 +51,7 @@ export interface ConstructorOptionsType {
     routerBasePath?: string;
     rootNode?: string;
     reduxMiddleware?: Middleware[];
+    basename?: string;
```
Comment (myNameIsDu) [inline packages/launcher/src/module/app.tsx:54]: 这个是不是重复了 和 routerBasePath

**Review by myNameIsDu — CHANGES_REQUESTED** [review]

File: packages/launcher/src/module/app.tsx (hunk 3 above)
Comment (bigbigDreamer) [reply packages/launcher/src/module/app.tsx:54]: delete

File: packages/launcher/src/module/app.tsx (hunk 2 above)
Comment (bigbigDreamer) [reply packages/launcher/src/module/app.tsx:92]: 我感觉问题不大，因为 就只实例化一次

**Review by myNameIsDu — APPROVED** [review]

**Pull request merged**